org.scala-ide.sdt.debug/src/org/scalaide/debug/internal/model/VirtualValue.scala

```diff
@@ -0,0 +1,47 @@
+package org.scalaide.debug.internal.model
+
+import org.eclipse.debug.core.model.IValue
+import org.scalaide.debug.internal.ScalaDebugger
+import org.scalaide.debug.internal.ScalaDebugPlugin
+import org.eclipse.debug.core.model.IVariable
+
+case class VirtualValue
+    (refTypeName: String, valueString: String, fields: Seq[IVariable] = Seq.empty)
+    (implicit debugTarget: ScalaDebugTarget)
+      extends ScalaDebugElement(debugTarget)
+      with IValue {
+
+  override def isAllocated(): Boolean = false
+
+  override def getReferenceTypeName(): String = refTypeName
+
+  override def getValueString(): String = valueString
+
+  def withFields(vars: IVariable*) =
+    VirtualValue(refTypeName, valueString, fields ++ vars.toSeq)
+
+  override def getVariables(): Array[IVariable] = fields.toArray
+
+  override def hasVariables(): Boolean = fields.isEmpty
```

Comment:
I think this would normally be `!fields.isEmpty`, right? (my mistake originally)